pkg/cmd/issue/edit/edit.go

```diff
@@ -141,75 +148,125 @@ func editRun(opts *EditOptions) error {
 	if err != nil {
 		return err
 	}
 
-	if opts.Interactive {
-		editorCommand, err := opts.DetermineEditor()
-		if err != nil {
-			return err
+	// Update all issues in parallel.
+	editedIssueChan := make(chan string, len(issues))
+	failedIssueChan := make(chan string, len(issues))
+	g := sync.WaitGroup{}
+
+	opts.IO.StartProgressIndicatorWithLabel(fmt.Sprintf("Updating %d issues", len(issues)))
+	for _, issue := range issues {
+		// Copy variables to capture in the go routine below.
+		editable := editable.Clone()
+
+		editable.Title.Default = issue.Title
+		editable.Body.Default = issue.Body
+		editable.Assignees.Default = issue.Assignees.Logins()
+		editable.Labels.Default = issue.Labels.Names()
+		editable.Projects.Default = append(issue.ProjectCards.ProjectNames(), issue.ProjectItems.ProjectTitles()...)
+		projectItems := map[string]string{}
+		for _, n := range issue.ProjectItems.Nodes {
+			projectItems[n.Project.ID] = n.ID
 		}
-		err = opts.EditFieldsSurvey(&editable, editorCommand)
-		if err != nil {
-			return err
+		editable.Projects.ProjectItems = projectItems
+		if issue.Milestone != nil {
+			editable.Milestone.Default = issue.Milestone.Title
+		}
+
+		// Allow interactive prompts for one issue; failed earlier if multiple issues specified.
+		if opts.Interactive {
+			editorCommand, err := opts.DetermineEditor()
+			if err != nil {
+				return err
+			}
+			err = opts.EditFieldsSurvey(&editable, editorCommand)
+			if err != nil {
+				return err
+			}
 		}
+
+		g.Add(1)
+		go func(issue *api.Issue) {
+			defer g.Done()
+
+			err := prShared.UpdateIssue(httpClient, repo, issue.ID, issue.IsPullRequest(), editable)
+			if err != nil {
+				failedIssueChan <- fmt.Sprintf("failed to update %s: %s", issue.URL, err)
+				return
+			}
+
+			editedIssueChan <- issue.URL
+		}(issue)
 	}
 
-	opts.IO.StartProgressIndicator()
-	err = prShared.UpdateIssue(httpClient, repo, issue.ID, issue.IsPullRequest(), editable)
+	g.Wait()
 	opts.IO.StopProgressIndicator()
-	if err != nil {
-		return err
+	close(editedIssueChan)
+	close(failedIssueChan)
+
+	// Print a sorted list of successfully edited issue URLs to stdout.
+	editedIssueURLs := make([]string, 0, len(issues))
+	for editedIssueURL := range editedIssueChan {
+		editedIssueURLs = append(editedIssueURLs, editedIssueURL)
+	}
+
+	sort.Strings(editedIssueURLs)
+	for _, editedIssueURL := range editedIssueURLs {
+		fmt.Fprintln(opts.IO.Out, editedIssueURL)
 	}
 
-	fmt.Fprintln(opts.IO.Out, issue.URL)
+	// Print a sorted list of failures to stderr.
+	failedIssueErrors := make([]string, 0, len(issues))
+	for failedIssueError := range failedIssueChan {
+		failedIssueErrors = append(failedIssueErrors, failedIssueError)
+	}
+
+	sort.Strings(failedIssueErrors)
+	for _, failedIssueError := range failedIssueErrors {
+		fmt.Fprintln(opts.IO.ErrOut, failedIssueError)
+	}
+
+	if len(failedIssueErrors) > 0 {
+		return fmt.Errorf("failed to update %d issue(s)", len(failedIssueErrors))
```

Comment:
I considered that, but I thought if anyone was ever checking stdout against some pattern, that would be a bit annoying since the text could change. But if that's not really much a concern, I will use `text.Pluralize`.

Reply:
I think it is okay to use `text.Pluralize`. My thought would be that the user should really be checking the exit code first and then if it is non-zero the user will likely need to investigate the output manually rather than trying to rely on pattern matching the output to determine what failed.